builtin/providers/vault/provider.go

```diff
@@ -122,6 +124,21 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		return nil, fmt.Errorf("failed to configure Vault API: %s", err)
 	}
 
+	token := d.Get("token").(string)
+	if token == "" {
+		// Use the vault CLI's token, if present.
+		homePath, err := homedir.Dir()
+		if err != nil {
+			return nil, fmt.Errorf("No vault token found: %s", err)
```

Comment:
Wouldn't this error be called if the homedir was unable to be found? The error text might need to be clarified.